coins/test_support.go

```diff
@@ -30,7 +30,7 @@ func StrToDecimal(amount string) *apd.Decimal {
 // StrToETHAsset converts a string into the EthAssetAmount. Pass nil for the
 // token if the asset is ETH.
 func StrToETHAsset(amount string, optionalToken *ERC20TokenInfo) EthAssetAmount {
-	return NewEthAssetAmount(StrToDecimal(amount), nil)
+	return NewEthAssetAmount(StrToDecimal(amount), optionalToken)
```

Comment:
Noticed the bug when it got flagged as an unused parameter by the linter. All callers are currently passing `nil`, so it didn't really matter.